### PR TITLE
lesson-3

### PR DESCRIPTION
**Pull request by Martochkaa**

### CODE REVIEW

File: src/components/Top/HeaderTop.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import TripHous from '../../img/icons/logo_vector.png';
+import night from '../../img/icons/Night.png';
+import account from '../../img/icons/Vector.png';
```
Comment (Dronom) [inline src/components/Top/HeaderTop.js]: need rename. like night-icon

File: src/components/Top/Top.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return (
+      <>
+      <div className="top-section" style={{ backgroundImage: `url('${background}')` }} >
+        <HeaderTop/>
+        <Filter setAvailableHotels={this.setAvailableHotels}/>
+      </div>
+      {this.state.isAvailableHotelsVisible ? 
+          <AvailableHotels hotelData={this.state.availableHotels} /> : 
+          <></>
+      }
```
Comment (Dronom) [inline src/components/Top/Top.js]: you can write it shorter
`{this.state.isAvailableHotelsVisible && 
          <AvailableHotels hotelData={this.state.availableHotels}/>
}`

File: src/components/filter/AvailableHotels.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    speed: 500,
+    slidesToShow: 4,
+    slidesToScroll: 4,
+    prevArrow: <BackArrow />,
+    nextArrow: <NextArrow />,
+  }
+  : {
+    slidesToShow: props.hotelData.length,
+    prevArrow: <></>,
+    nextArrow: <></>,
```
Comment (Dronom) [inline src/components/filter/AvailableHotels.js]: use null for that

File: src/components/filter/CityFilter.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    return (
+        <>
+        <div className="city-input-container col-md-12">
+            <input id="citySearchInput" className="city-input input-text col-md-12" type="text" name="search" required autoComplete="off"
+                onChange={filter}
+            />
+            <label className="city-input-label">Your destination or hotel name</label>
+        </div>
+        </>
```
Comment (Dronom) [inline src/components/filter/CityFilter.js]: unnecessary <></> wrapper

File: src/components/filter/Filter.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    <input onClick={this.clickRoomFilter} className="guest-input input-text col-md-12" id="guestInput" type="text" placeholder="2 Adults — 0 Children — 1 Room" autoComplete="off" ></input>
+                    <button onClick={this.showAvailableHotels} className="search-button col-md-12" id="searchHotel" type="submit" form="mainForm">Search</button>
+                </div>
+                <div className="apple-google">
+                <a href="https://play.google.com"><img className="google-img" src={google} alt="google"/></a>
+                <a href="https://www.apple.com/ru/app-store/"><img className="apple-img" src={app} alt="app"/></a>
+                </div>
+                <div className={`filter-form${this.state.isRoomFilterHide ? " hide" : ""}`} id="filterForm">
+                    <RoomFilter/>
+                </div>
```
Comment (Dronom) [inline src/components/filter/Filter.js:63]: the class changes, but the filter is never shown
image

File: src/components/filter/RoomFilter.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React  from 'react';
+import FilterItem from './RoomFilterItem';
+import '../../style/TopStyle/Form.css';
+
+function plusParent() {
+
+}
```
Comment (Dronom) [inline src/components/filter/RoomFilter.js]: remove

File: src/components/filter/RoomFilterItem.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+function FilterItem(props) {
+
+    const {plusEvent, minusEvent, count} = props;
+
+    return (
+        <div className="counter-item">
+            <span className="counter-label">Adult</span>
+            <div className="buttons-input-wrapper"> 
+                <button onClick={minusEvent} type="button" className="counter-plus-minus-gray counter-btn" id="btnCounterAdultMinus" data-direction="minus">-</button>
+                <input className="counter-current-value" type="text" id="counterAdult" value={count} form="mainForm"></input>
```
Comment (Dronom) [inline src/components/filter/RoomFilterItem.js:14]: image

File: src/components/filter/Filter.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+            <article>
+                <div className="city-date-form" autoFocus>
+                    <div className="city-input-container col-md-12">
+                        <input id="citySearchInput" className="city-input input-text col-md-12" type="text" name="search" required autoComplete="off" defaultValue=""
+                            onChange={this.changeCityFilter}
+                        />
+                        <label className="city-input-label">Your destination or hotel name</label>
+                    </div>
+                    <DateFilter/>
+                    <input onClick={this.clickRoomFilter} className="guest-input input-text col-md-12" id="guestInput" type="text" placeholder="2 Adults — 0 Children — 1 Room" autoComplete="off" ></input>
```
Comment (Dronom) [inline src/components/filter/Filter.js:54]: I'm not sure then it's a good idea to use onclick on input without onchange. If you are using onclick, then the button is better suited for this.

**Review by Dronom — CHANGES_REQUESTED** [review]
styles need a little tweak
image

the use of PropTypes is commendable, but pay attention then to the warnings that appear
image

File: src/components/BackArrow.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
 
 const BackArrow = ({onClick}) => {
     return (
-        <div className="BackArrow" onClick={onClick} >
+      <div className="BackArrow" onClick={onClick} >
```
Comment (Dronom) [inline src/components/BackArrow.js:8]: make sense to do it as a button (the same for NextArrow)

**Review by Dronom — COMMENTED** [review]
Looks better, take a look at remark

File: src/components/BackArrow.js (hunk 9 above)
Comment (Martochkaa) [reply src/components/BackArrow.js:8]: BUGS FIXED IN PULL REQUEST LESSON-4

**Pull request merged**